### PR TITLE
Adds MetaStation Access Helpers

### PR DESCRIPTION
**Pull request by Son-of-Space**
## About The Pull Request

This PR adds a few types of mapping helpers and fully maps MetaStation to utilize them. Many access requirements were standardized to get rid of snowflake access needs and the following was standardized for maintenance:
1. Departments have access to their general area of maintenance.
2. People with Maintenance access have access to all maintenance
3. Maintenance doors have access helpers to allow someone to get out in the event of being trapped.

## Why It's Good For The Game
Standardizes access requirements for the map and reduces changes for inconsistent mapping to happen by use of mapping helpers. Makes access requirements easier to audit.

## Changelog

:cl:
fix: Fixed several inconsistent or niche access requirements on MetaStation
qol: Replaced all access requirement vars on doors with mapping helpers on MetaStation
/:cl:

### CODE REVIEW

**Review by OperativeLyn — CHANGES_REQUESTED** [review]
You missed CE's office, I reckon.
image

File: code/modules/mapping/access_helpers.dm [hunk 1]
```diff
@@ -97,6 +101,11 @@
 	access_list += ACCESS_MAINT_TUNNELS
 	return access_list
 
+/obj/effect/mapping_helpers/airlock/access/any/engineering/maintenance/departmental/get_access()
+	var/list/access_list = ..()
+	access_list += list(ACCESS_ENGINE,ACCESS_MAINT_TUNNELS)
```
Comment (san7890) [inline code/modules/mapping/access_helpers.dm:106]:
```suggestion
	access_list += list(ACCESS_ENGINE, ACCESS_MAINT_TUNNELS)
```

File: code/modules/mapping/access_helpers.dm [hunk 2]
```diff
@@ -68,6 +68,10 @@
 	access_list += ACCESS_CAPTAIN
 	return access_list
 
+/obj/effect/mapping_helpers/airlock/access/any/command/maintenance/get_access()
+	var/list/access_list = ..()
+	access_list += list(ACCESS_HEADS, ACCESS_MAINT_TUNNELS)
+	return access_list
 // -------------------- Engineering access helpers
```
Comment (san7890) [inline code/modules/mapping/access_helpers.dm:75]:
```suggestion
	return access_list

// -------------------- Engineering access helpers
```
Newline for separation.

**Review by san7890 — CHANGES_REQUESTED** [review]
I'll look through the map itself soon enough, but I took a precursory look at the pathing you introduced.

**Review by Maurukas — APPROVED** [review]
Everything looks correct here, in terms of both the added helpers and their addition to Metastation. Testing a half dozen jobs on a local server also comes back with everything behaving itself.

This is a fairly big access overhaul, so I'd appreciate you keeping an eye out for any issues that pop up over the next few days.

**Comment by Son-of-Space** [comment]
Will do

**Pull request merged**